src/notification/notification.service.ts

```diff
@@ -0,0 +1,37 @@
+import { EmailService } from "../email/email.service";
+import { Inject, Injectable } from "@nestjs/common";
+import { TemplateService } from "../template/template.service";
+import { OnEvent } from "@nestjs/event-emitter";
+import { ConfigService } from "@nestjs/config";
+import { Email } from "../email/dto/email.dto";
+
+@Injectable()
+export class NotificationService {
+  constructor(
+    @Inject("TemplateService")
+    private readonly templateService: TemplateService,
+    @Inject("EmailService") private readonly emailService: EmailService,
+    private readonly configService: ConfigService
+  ) {}
+
+  @OnEvent("otp.login.email.created")
+  async onLoginOtpEmailCreated({
+    otpCode,
+    userEmailAddress,
+  }: {
+    otpCode: string;
+    userEmailAddress: string;
+  }): Promise<void> {
+    const template = this.templateService.getLoginOtpEmailTemplate("en");
+    const html = this.templateService.injectData(template, {
+      otpCode,
+    });
+    const email = {
```

Comment:
Create a private method named sendEmailAsync by getting content and subject as parameters.